holoviews/plotting/mpl/raster.py

```diff
@@ -141,6 +153,13 @@ class QuadMeshPlot(ColorbarPlot):
     def get_data(self, element, ranges, style):
         zdata = element.dimension_values(2, flat=False)
         data = np.ma.array(zdata, mask=np.logical_not(np.isfinite(zdata)))
+
+        plot_opts = element.opts.get('plot', 'matplotlib')
+        nodata = plot_opts.kwargs.get('nodata')
+        if nodata is not None:
+            data = data if (data.dtype.kind  == 'f') else data.astype(np.float64)
+            data[data == nodata] = np.NaN
```

Comment:
Same as above; docstring indicates that `nodata` applies only to integer data, but here it's applied to all data.

Reply:
Resolved in f146bd3